MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java

```diff
@@ -1286,6 +1286,14 @@ private void onLocationLayerStop() {
     }
 
     stopPulsingLocationCircle();
+
+    //Record the current layer location so that location layer animation continues from this point after change style.
+    LatLng currentLayerLocation = locationAnimatorCoordinator.getPreviousLayerLatLng();
+    if (currentLayerLocation != null && lastLocation != null) {
+      lastLocation.setLatitude(currentLayerLocation.getLatitude());
+      lastLocation.setLongitude(currentLayerLocation.getLongitude());
```

Comment:
The `lastLocation` field is also used as a return value for the `getLastLocation` that can be called from the outside and should return the true value, not a place where the animation stopped.